File: services/store/store.py
"""item 本地持久化（JSON 文件）。

M0 用 JSON 全量读写（DEC-STORE）。零依赖、可读可导出、离线。
文件损坏 → 读空兜底（不崩）。charter 第1条数据属用户：开放格式。
"""
from __future__ import annotations

import json
from dataclasses import asdict
from pathlib import Path

from services.capture.classify import Item


class ItemStore:
    def __init__(self, path: str | Path):
        self._path = Path(path)

    def add(self, item: Item) -> None:
        items = self.list()
        items.append(item)
        self._save(items)

    def list(self) -> list[Item]:
        if not self._path.exists():
            return []
        try:
            raw = json.loads(self._path.read_text(encoding="utf-8"))
        except (json.JSONDecodeError, OSError):
            return []  # 文件损坏兜底：读空，不崩
        return [Item(**d) for d in raw]

    def clear(self) -> None:
        self._save([])

    def _save(self, items: list[Item]) -> None:
        self._path.parent.mkdir(parents=True, exist_ok=True)
        self._path.write_text(
            json.dumps([asdict(i) for i in items], ensure_ascii=False, indent=2),
            encoding="utf-8",
        )

File: apps/mobile/lib/store/store.dart
// 端侧本地存储（Dart，JSON 文件 + path_provider）。
// 复刻 services/store/store.py；契约 task-store。离线、可导出。
import 'dart:convert';
import 'dart:io';

import 'package:kite_mobile/capture/classify.dart';

class ItemStore {
  final String path;
  ItemStore(this.path);

  Future<void> add(Item item) async {
    final items = await list();
    items.add(item);
    await _save(items);
  }

  Future<List<Item>> list() async {
    final f = File(path);
    if (!await f.exists()) return [];
    try {
      final raw = jsonDecode(await f.readAsString()) as List;
      return raw.map((d) => Item.fromJson(d as Map<String, dynamic>)).toList();
    } catch (_) {
      return []; // 文件损坏兜底：读空
    }
  }

  Future<void> clear() async => _save([]);

  Future<void> _save(List<Item> items) async {
    final f = File(path);
    await f.parent.create(recursive: true);
    await f.writeAsString(
      const JsonEncoder.withIndent('  ').convert(items.map((i) => i.toJson()).toList()),
    );
  }
}
